discovery/discovery_test.go

```diff
@@ -222,3 +328,187 @@ func TestNewOrchestratorPoolCache_GivenListOfOrchs_CreatesPoolCacheCorrectly(t *
 		assert.Equal(uri.String(), expected[i])
 	}
 }
+
+func TestNewOrchestratorPoolWithPred_TestPredicate(t *testing.T) {
+	pred := func(info *net.OrchestratorInfo) bool {
+		server.BroadcastPrice.PriceMutex.RLock()
+		defer server.BroadcastPrice.PriceMutex.RUnlock()
+		if server.BroadcastPrice.Price != nil {
+			return big.NewRat(info.PriceInfo.PricePerUnit, info.PriceInfo.PixelsPerUnit).Cmp(server.BroadcastPrice.Price) <= 0
+		}
+		return true
+	}
+
+	addresses := []string{}
+	for i := 0; i < 50; i++ {
+		addresses = append(addresses, "https://127.0.0.1:8936")
+	}
+	orchestrators := StubOrchestrators(addresses)
+
+	node, _ := core.NewLivepeerNode(nil, "", nil)
+	node.Eth = &eth.StubClient{Orchestrators: orchestrators}
+
+	pool := NewOrchestratorPoolWithPred(node, addresses, pred)
+
+	oInfo := &net.OrchestratorInfo{
+		PriceInfo: &net.PriceInfo{
+			PricePerUnit:  5,
+			PixelsPerUnit: 1,
+		},
+	}
+
+	// server.BroadcastPrice.Price not yet set, predicate should return true
+	assert.True(t, pool.pred(oInfo))
+
+	// Set MaxBroadcastPrice higher than PriceInfo , should return true
+	server.BroadcastPrice.Price = big.NewRat(10, 1)
+	assert.True(t, pool.pred(oInfo))
+
+	// Set MaxBroadcastPrice lower than PriceInfo, should return false
+	server.BroadcastPrice.Price = big.NewRat(1, 1)
+	assert.False(t, pool.pred(oInfo))
+}
+
+func TestCachedPool_AllOrchestratorsTooExpensive_ReturnsEmptyList(t *testing.T) {
```

Comment:
So, we have a test for the following:

- All Os too expensive
- Max broadcast price is not set

It would be nice to also have a test that checks for N out of 50 Os meeting the max price requirements.

Reply:
Yeah I realized that but had no immeadiate idea on how to implement it cleanly. Any suggestions?

Reply:
The stubbed [serverGetOrchInfo()](https://github.com/livepeer/go-livepeer/pull/933/files#diff-13f8a7f541eee06b276d455b486bb2efR381) function in the tests accepts a `orchestratorServer *url.URL` argument. We could modify the stubbed function to return an unacceptable `PriceInfo` message for different `orchestratorServer` values.

Ex.

```
// Initialize with URLs of Os that should return an unacceptable PriceInfo message
unacceptablePriceURLs := ...

serverGetOrchInfo = func(ctx context.Context, bcast server.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
    ...

    if _, ok := unacceptablePriceURLs[orchestratorServer]; ok {
        // Return an unacceptable PriceInfo message
    }

    // Return an acceptable PriceInfo message
}
```